src/sidebar/store/modules/annotations.js

```diff
@@ -441,6 +439,31 @@ const isWaitingToAnchorAnnotations = createSelector(
   annotations => annotations.some(metadata.isWaitingToAnchor)
 );
 
+/**
+ * @typedef AnnotationsStore
+ *
+ * // actions
+ * @prop {typeof addAnnotations} addAnnotations
+ * @prop {typeof clearAnnotations} clearAnnotations
+ * @prop {typeof hideAnnotation} hideAnnotation
+ * @prop {typeof removeAnnotations} removeAnnotations
+ * @prop {typeof updateAnchorStatus} updateAnchorStatus
+ * @prop {typeof updateFlagStatus} updateFlagStatus
+ * @prop {typeof unhideAnnotation} unhideAnnotation
+ *
+ * // selectors
+ * @prop {() => number} annotationCount
+ * @prop {(a: string) => boolean} annotationExists
+ * @prop {(a: string) => Annotation} findAnnotationByID
+ * @prop {(a: string[]) => string[]} findIDsForTags
```

Comment:
I guess my question is are these function param types used for readability or just type compliance. Because one can always (and imo) should look up the proper function definition. However I could see this going both ways... the actual function definition is difference because it takes a `state` prop which the user should ignore.

hmmm... maybe I'm leaning towards using non-generic names. My fear is an unnecessary upkeep /parity down the road.